### PR TITLE
perf(http): remove IE special status handling

### PR DESCRIPTION
**Pull request by diicar**
## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## What is the current behavior?
Contains a workaround for IE

## What is the new behavior?
Removed IE special Case handling

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by diicar** [comment]
cla signed

**Comment by dylhunn** [comment]
It looks like CircleCI failed to run due to a rebase conflict. Can you rebase on master and push again?

**Review by dylhunn — CHANGES_REQUESTED** [review]
(Needs a rebase and fresh CI run)

**Comment by diicar** [comment]
conflicts are now resolved.
I updated the description by moving up because of **ci/circleci: lint** . 
Can you please rerun  **ci/circleci: lint**.
Thank you.

File: packages/common/http/src/xhr.ts [hunk 1]
```diff
@@ -49,11 +49,13 @@ export class HttpXhrBackend implements HttpBackend {
    * @returns An observable of the response events.
    */
   handle(req: HttpRequest<any>): Observable<HttpEvent<any>> {
-    // Quick check to give a better error message when a user attempts to use
-    // HttpClient.jsonp() without installing the HttpClientJsonpModule
-    if (req.method === 'JSONP') {
-      throw new Error(
-          `Attempted to construct Jsonp request without HttpClientJsonpModule installed.`);
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
```
Comment (alxhub) [inline packages/common/http/src/xhr.ts:52]: This seems like a candidate for @AndrewKushnir's improvements to `RuntimeError` - we could use this to make the actual error message tree-shakable while still catching the issue even in production mode.

Comment (AndrewKushnir) [reply packages/common/http/src/xhr.ts:52]: @diicar as Alex mentioned above, we'd like to keep the check, but improve the code by leveraging the `RuntimeError` class and wrapping the error message (which contributes to the payload size the most) in `ngDevMode` check, [see here for an example](https://github.com/angular/angular/commit/67df935b411ca5c79b33dd13c93933c41e8e3ae4#diff-61abe7c733b66d542f4f5bd4365382ab7c6a7a55327e3b7cb09d46b6c376b561R207).

For this PR though, my proposal is to revert that change (just keep the IE-related cleanup) and create a separate PR with the refactoring to the error message. Please let me know (as a reply here) if you have any questions.

Comment (diicar) [reply packages/common/http/src/xhr.ts]: Thank you guys. 
Reverted the changes.

Comment (diicar) [reply packages/common/http/src/xhr.ts]: updated the description and squashed the commits into one.

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:426256229:BASE:426256276:1643931953310:5fd8bde8).

**Review by AndrewKushnir — APPROVED** [review]

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-http

**Comment by AndrewKushnir** [comment]
@dylhunn FYI the TGP is green, this PR is ready to go.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 28393031b10f4dbefab7587c19641e49cb820785.

**Pull request closed without merging**